core/domain/skill_services.py

```diff
@@ -441,6 +441,53 @@ def get_all_topic_assignments_for_skill(skill_id):
     return topic_assignments
 
 
+def replace_skill_id_in_all_topics(user_id, old_skill_id, new_skill_id):
+    """Replaces the old skill id with the new one in all the associated topics.
+
+    Args:
+        user_id: str. The unique user ID of the user.
+        old_skill_id: str. The old skill id.
+        new_skill_id: str. The new skill id.
+    """
+    all_topics = topic_fetchers.get_all_topics()
+    for topic in all_topics:
+        change_list = []
+        if old_skill_id in topic.get_all_skill_ids():
+            if new_skill_id in topic.get_all_skill_ids():
+                raise Exception(
+                    'Found topic \'%s\' contains the two skills to be merged. '
+                    'Please unassign one of these skills from topic '
+                    'and retry this operation.' % topic.name)
+            if old_skill_id in topic.uncategorized_skill_ids:
+                change_list.extend([topic_domain.TopicChange({
+                    'cmd': 'remove_uncategorized_skill_id',
+                    'uncategorized_skill_id': old_skill_id
+                }), topic_domain.TopicChange({
+                    'cmd': 'add_uncategorized_skill_id',
+                    'new_uncategorized_skill_id': new_skill_id
+                })])
```

Comment:
Add this below the subtopic loop below, since if a skill ID is removed from the subtopic, it goes to uncategorized, so the old skill ID will remain in uncategorized in the loop below.

Reply:
Ah got it. I've kept this here since I'm doing a `remove_uncategorized_skill_id` followed by a `add_uncategorized_skill_id` for the new skill id. But I've now added `remove_uncategorized_skill_id` for the old skill id in the block handling the logic for subtopics below. PTAL!